### PR TITLE
Fix error messages from OIDC config parsing

### PR DESCRIPTION
**Pull request by richvdh**
#9110 reshuffled the logic for parsing OIDC config, but didn't update the error messages to match.

### CODE REVIEW

File: synapse/config/oidc_config.py [hunk 1]
```diff
@@ -280,8 +280,8 @@ def _parse_oidc_provider_configs(config: JsonDict) -> Iterable["OidcProviderConf
     """
     validate_config(MAIN_CONFIG_SCHEMA, config, ())
 
-    for p in config.get("oidc_providers") or []:
-        yield _parse_oidc_config_dict(p)
+    for i, p in enumerate(config.get("oidc_providers")) or []:
```
Comment (clokep) [inline synapse/config/oidc_config.py:283]: I suspect this should be:
```suggestion
    for i, p in enumerate(config.get("oidc_providers") or []):
```

This should handle bogus values better (e.g. `oidc_providers: null` or something.

**Review by clokep — COMMENTED** [review]
Looks like mypy is unhappy too.

File: synapse/config/oidc_config.py (hunk 1 above)
Comment (richvdh) [reply synapse/config/oidc_config.py]: :facepalm: thanks

**Review by clokep — APPROVED** [review]
LGTM.

**Pull request merged**